gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/portal_page/use_case/ListPortalNavigationItemsUseCase.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.core.portal_page.use_case;
+
+import io.gravitee.apim.core.UseCase;
+import io.gravitee.apim.core.portal_page.model.PortalArea;
+import io.gravitee.apim.core.portal_page.model.PortalNavigationFolder;
+import io.gravitee.apim.core.portal_page.model.PortalNavigationItem;
+import io.gravitee.apim.core.portal_page.model.PortalNavigationItemId;
+import io.gravitee.apim.core.portal_page.query_service.PortalNavigationItemsQueryService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+
+@UseCase
+@RequiredArgsConstructor
+public class ListPortalNavigationItemsUseCase {
+
+    private final PortalNavigationItemsQueryService queryService;
+
+    public Output execute(Input input) {
+        var directItems = input
+            .parentId()
+            .map(parentId -> queryService.findByParentIdAndEnvironmentId(input.environmentId(), parentId))
+            .orElse(queryService.findTopLevelItemsByEnvironmentId(input.environmentId(), input.portalArea()));
```

Comment:
Should we explicitly mention `area`?
```suggestion
            .map(parentId -> queryService.findByParentIdAndEnvironmentIdAndArea(input.environmentId(), parentId, input.portalArea()))
            .orElse(queryService.findTopLevelItemsByEnvironmentIdAndArea(input.environmentId(), input.portalArea()));
```

Reply:
If we would explicitly mention area, it would mean that single parent can have children across different areas, which isn't true. And also it would be an additional step in query execution plan. I am not a fan of this idea.

Reply:
For me it's more a check that the requested parentId is effectively part of that environment and that area

Reply:
I don't think we really need that check...

Reply:
ok for the first one, but at least do you agree on the second method renaming ?

Reply:
Yes, yes. I done that already.